### ISSUE TITLE
[azure] blobfile fails to access container I have "Storage Blob Data Contributor" permissions to, through a group

### ISSUE THREAD

**Issue by hponde** [christopher-hesse/blobfile#110]
I'm part of a Group G that has access to storage account A, with the "Storage Blob Data Contributor" role.

I'm able to run `az storage blob list --account-name A -c [some_container]` with the az CLI, but blobfile fails to load:

`blobfile.BlobFile("https://A.blob.core.windows.net/some_container/...")`

fails with `blobfile.common.Error: Storage account ID not found for storage account: 'A'`

**Comment by hponde** [comment]
It seems that if instead group G is added as `Owner` then this code just works. This might be the best workaround for now.

**Comment by christopher-hesse** [comment]
@hponde What error do you get with the master version?  Also does it work if you do `az storage blob list --account-name A -c [some_container] --auth-mode=login`?  In the past, people have often messed up the permissions so that the azure cli worked with auth mode key but not auth mode login.

**Comment by hponde** [comment]
I even have `AZURE_STORAGE_AUTH_MODE=login` set locally so it works for `login`. And just to check I tried specifying `--auth-mode=login` and it works, while `--auth-mode=key` does not.

This was on version `0.17.0` which is latest.

We have gotten around this by giving `Owner` role to the group and not the `blobfile` call does work, so it's a bit harder to reproduce (I could try it tomorrow).

The error is that it thinks the storage account doesn't exist: `blobfile.common.Error: Storage account ID not found for storage account: 'A'` (I replaced the actual name with `A` here)
